C7/MapView.cs

```diff
@@ -33,6 +33,29 @@ public class TerrainLayer : LooseLayer {
 	// A triple sheet is a sprite sheet containing sprites for three different terrain types including transitions between.
 	private List<ImageTexture> tripleSheets;
 
+	// TileToDraw stores the arguments passed to drawObject so the draws can be sorted by texture before being submitted. This significantly
+	// reduces the number of draw calls Godot must generate (1483 to 312 when fully zoomed out on our test map) and modestly improves framerate
+	// (by about 14% on my system).
+	private class TileToDraw : IComparable<TileToDraw>
+	{
+		public Tile tile;
+		public Vector2 tileCenter;
+
+		public TileToDraw(Tile tile, Vector2 tileCenter)
+		{
+			this.tile = tile;
+			this.tileCenter = tileCenter;
+		}
+
+		public int CompareTo(TileToDraw other)
+		{
+			// "other" might be null, in which case we should return that this object is greater, according to an example on MSDN
+			return (other != null) ? this.tile.ExtraInfo.BaseTerrainFileID.CompareTo(other.tile.ExtraInfo.BaseTerrainFileID) : 1;
```

Comment:
quick tip, this can also be written as:
`return this.tile.ExtraInfo.BaseTerrainFileID.CompareTo(other?.tile.ExtraInfo.BaseTerrainFileID);`
(notice the ?.) since CompareTo(null) returns a positive value